content/en/platform/corda/5.0-dev-preview-2/developing/api/amqp-serialization-enum-evolution.md

```diff
@@ -0,0 +1,337 @@
+---
+date: '2022-09-21T14:27:00+01:00'
+title: "Enum Evolution"
+menu:
+  corda-5-dev-preview:
+    identifier: corda-5-dev-preview-api-serialisation-enum-evolution
+    parent: corda-5-dev-preview-api-serialisation
+    weight: 8000
+section_menu: corda-5-dev-preview
+---
+
+
+# Enum Evolution
+
+In the continued development of a CorDapp an enumerated type that was fit for purpose at one time may
+require changing. Normally, this would be problematic as anything serialised (and kept in a vault) would
+run the risk of being unable to be deserialized in the future or older versions of the app still alive
+within a compatibility zone may fail to deserialize a message.
+
+To facilitate backward and forward support for alterations to enumerated types Corda’s serialization
+framework supports the evolution of such types through a well defined framework that allows different
+versions to interoperate with serialised versions of an enumeration of differing versions.
+
+This is achieved through the use of certain annotations. Whenever a change is made, an annotation
+capturing the change must be added (whilst it can be omitted any interoperability will be lost). Corda
+supports two modifications to enumerated types, adding new constants, and renaming existing constants
+
+{{< warning >}}
+Once added, evolution annotations MUST NEVER be removed from a class, doing so will break
+both forward and backward compatibility for this version of the class and any version moving
+forward.
+{{< /warning >}}
+
+## The Purpose of Annotating Changes
+
+The biggest hurdle to allowing enum constants to be changed is that there will exist instances of those
+classes, either serialized in a vault or on nodes with the old, unmodified, version of the class that we
+must be able to inter-operate with. Thus if a received data structure references an enum assigned a constant
+value that doesn’t exist on the running JVM, a solution is needed.
+
+For this, we use the annotations to allow developers to express their backward compatible intentions.
+
+In the case of renaming constants this is somewhat obvious, the deserializing node will simply treat any
+constants it doesn’t understand as their “old” values, i.e. those values that it currently knows about.
+
+In the case of adding new constants the developer must chose which constant (that existed *before* adding
+the new one) a deserializing system should treat any instances of the new one as.
+
+{{< note >}}
+Ultimately, this may mean some design compromises are required. If an enumeration is
```

Comment:
Hard to follow..
Maybe:
Ultimately, this may mean some design compromises are required. If you expect that an enumeration will be extended often and no sensible defaults exist, including a constant
in the original version of the class that all new additions can default to, may make sense.

Reply:
Done